domain/agentbinary/state/types.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+// architectureRecord represents a architecture entry in the database.
+type architectureRecord struct {
+	ID   int    `db:"id"`
+	Name string `db:"name"`
+}
+
+// objectStoreMeta represents the object store metadata in the database
+// and exists is used to check if the object store metadata exists.
+type objectStoreMeta struct {
+	Exists bool   `db:"exists"`
+	UUID   string `db:"uuid"`
```

Comment:
Do we need all these fields?

Reply:
Done & Done

Reply:
I disagree here.
If we don't use them, they should be removed.
If they are used for tests only, they should be in the test file.